CI - File uploading fails occasionally on Windows with SignatureDoesNotMatch errors

Noticing some failures during the `Upload artifacts` step every once in awhile with:
`An error occurred (SignatureDoesNotMatch) when calling the PutObject operation: The request signature we calculated does not match the signature you provided. Check your key and signing method.`

A recent build:
https://github.com/ROCm/TheRock/actions/runs/15760354406/job/44425326792#step:19:71

It seems other targets build and upload okay in this workflow.

Another failure: https://github.com/ROCm/TheRock/actions/runs/19203073956/job/54894316318#step:18:73

```
Run aws s3 cp B:\build/packages/dist/ s3://therock-nightly-python/v2-staging/gfx110X-dgpu/ \
upload failed: B:\build\packages\dist\rocm-7.10.0a20251109.tar.gz to s3://therock-nightly-python/v2-staging/gfx110X-dgpu/rocm-7.10.0a20251109.tar.gz An error occurred (SignatureDoesNotMatch) when calling the PutObject operation: The request signature we calculated does not match the signature you provided. Check your key and signing method.
Completed 15.4 KiB/2.1 GiB (14.3 KiB/s) with 4 file(s) remaining
upload failed: B:\build\packages\dist\rocm_sdk_libraries_gfx110x_dgpu-7.10.0a20251109-py3-none-win_amd64.whl to s3://therock-nightly-python/v2-staging/gfx110X-dgpu/rocm_sdk_libraries_gfx110x_dgpu-7.10.0a20251109-py3-none-win_amd64.whl An error occurred (SignatureDoesNotMatch) when calling the CreateMultipartUpload operation: The request signature we calculated does not match the signature you provided. Check your key and signing method.
Completed 15.4 KiB/2.1 GiB (14.3 KiB/s) with 3 file(s) remaining
upload failed: B:\build\packages\dist\rocm_sdk_devel-7.10.0a20251109-py3-none-win_amd64.whl to s3://therock-nightly-python/v2-staging/gfx110X-dgpu/rocm_sdk_devel-7.10.0a20251109-py3-none-win_amd64.whl An error occurred (SignatureDoesNotMatch) when calling the CreateMultipartUpload operation: The request signature we calculated does not match the signature you provided. Check your key and signing method.
Completed 189.4 MiB/2.1 GiB (14.3 KiB/s) with 2 file(s) remaining
upload failed: B:\build\packages\dist\rocm_sdk_core-7.10.0a20251109-py3-none-win_amd64.whl to s3://therock-nightly-python/v2-staging/gfx110X-dgpu/rocm_sdk_core-7.10.0a20251109-py3-none-win_amd64.whl An error occurred (SignatureDoesNotMatch) when calling the CreateMultipartUpload operation: The request signature we calculated does not match the signature you provided. Check your key and signing method.
Completed 1.5 GiB/2.1 GiB (14.3 KiB/s) with 1 file(s) remaining
Error: Process completed with exit code 1.
Run aws s3 cp B:\build/packages/dist/ s3://therock-nightly-python/v2-staging/gfx110X-dgpu/ \
upload failed: B:\build\packages\dist\rocm-7.10.0a20251109.tar.gz to s3://therock-nightly-python/v2-staging/gfx110X-dgpu/rocm-7.10.0a20251109.tar.gz An error occurred (SignatureDoesNotMatch) when calling the PutObject operation: The request signature we calculated does not match the signature you provided. Check your key and signing method.
Completed 15.4 KiB/2.1 GiB (14.3 KiB/s) with 4 file(s) remaining
```

https://github.com/ROCm/TheRock/commit/54b95fe15fc5552ee378bf0ac86495a83667187a added `--region us-east-2` to a single upload command hoping it might help. There are multiple other places where we have upload commands that can fail. We also still want to switch from the `aws` CLI to boto3 (the AWS SDK for Python), maybe that will help (or have different errors).

--- a/.github/workflows/build_windows_artifacts.yml
+++ b/.github/workflows/build_windows_artifacts.yml
@@ -186,6 +186,7 @@ jobs:
         with:
           aws-region: us-east-2
           role-to-assume: arn:aws:iam::692859939525:role/therock-ci
+          special-characters-workaround: true
 
       - name: Post Build Upload
         if: always()

--- a/.github/workflows/build_windows_pytorch_wheels.yml
+++ b/.github/workflows/build_windows_pytorch_wheels.yml
@@ -229,6 +229,7 @@ jobs:
         with:
           aws-region: us-east-2
           role-to-assume: arn:aws:iam::692859939525:role/therock-${{ inputs.release_type }}-releases
+          special-characters-workaround: true
 
       - name: Upload wheels to S3 staging
         if: ${{ github.repository_owner == 'ROCm' }}
@@ -308,6 +309,7 @@ jobs:
         with:
           aws-region: us-east-2
           role-to-assume: arn:aws:iam::692859939525:role/therock-${{ inputs.release_type }}-releases
+          special-characters-workaround: true
 
       - name: Determine upload flag
         env:

--- a/.github/workflows/release_windows_packages.yml
+++ b/.github/workflows/release_windows_packages.yml
@@ -274,6 +274,7 @@ jobs:
         with:
           aws-region: us-east-2
           role-to-assume: arn:aws:iam::692859939525:role/therock-${{ env.RELEASE_TYPE }}
+          special-characters-workaround: true
 
       # TODO: upload logs even if the build fails (separate from artifact upload?)
       - name: Post Build Upload

